iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/helpers/IntegrationTest.java

```diff
@@ -27,9 +27,14 @@ protected void finished(Description description)
         }
     };
 
+    @Rule
+    public final ConditionalIgnoreRule mConditionalIgnore = new ConditionalIgnoreRule();
+
     public static final int E2E_TEST_TIMEOUT_MS = 15 * 60 * 1000;
 
     //This timeout applies to all individual tests in classes that inherit from this class
     @Rule
     public Timeout timeout = new Timeout(E2E_TEST_TIMEOUT_MS);
+
+    public static boolean isBasicTierHub;
```

Comment:
I'm not convinced this belongs here. This class is supposed to be the general concept of an integration test. Not all integration tests will care if it is a basic tier iot hub or not.

Reply:
isBasicTierHub is used in all runner classes to determine which set of the tests to run...device vs module, and features.  Please suggest where you would like to declare it.

---
In reply to: [274587191](https://github.com/Azure/azure-iot-sdk-java/pull/485#discussion_r274587191) [](ancestors = 274587191)

Reply:
within each of the common classes where it is relevant, just like all the other environment variables. SendMessagesCommon, ReceiveMessagesCommon, etc.

---
In reply to: [274684601](https://github.com/Azure/azure-iot-sdk-java/pull/485#discussion_r274684601) [](ancestors = 274684601,274587191)